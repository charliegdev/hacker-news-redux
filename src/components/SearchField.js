import React, { Component } from "react";
import PropTypes from "prop-types";

class SearchField extends Component {
  constructor(props) {
    super(props);

    this.state = { searchValue: "" };

    this.onSearchFieldChange = this.onSearchFieldChange.bind(this);
  }

  onSearchFieldChange(event) {
    const newSearchValue = event.target.value;
    this.setState({ searchValue: newSearchValue });
    this.props.onChangeFunc(newSearchValue);
  }

  render() {
    const { onSubmitFunc, children } = this.props;
    return (
      <div>
        <p>
          <button
            className="btn btn-primary"
            type="button"
            data-toggle="collapse"
            data-target="#collapseExample"
            aria-expanded="false"
            aria-controls="collapseExample"
          >
            {children}
          </button>
        </p>
        <div className="collapse" id="collapseExample">
          <div className="card card-body">
            <form onSubmit={onSubmitFunc} onChange={this.onSearchFieldChange}>
              <div className="form-group">
                <label htmlFor="searchField">Search a Title</label>
                <input
                  type="text"
                  className="form-control"
                  id="searchField"
                  placeholder="Enter part of a title"
                  value={this.state.searchValue}
                />
                <small id="searchFieldHelp" className="form-text text-muted">
                  Searching for locally cached news
                </small>
              </div>
            </form>
          </div>
        </div>
        <br />
      </div>
    );
  }
}

SearchField.propTypes = {
  onChangeFunc: PropTypes.func.isRequired,
  onSubmitFunc: PropTypes.func.isRequired
};

export default SearchField;
