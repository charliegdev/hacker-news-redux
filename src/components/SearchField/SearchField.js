import React, { Component } from "react";
import PropTypes from "prop-types";

class SearchField extends Component {
  componentDidMount() {
    if (this.input) this.input.focus();
  }

  render() {
    const { onSubmitFunc, onChangeFunc, searchValue, children = "Search" } = this.props;
    return (
      <div>
        <div className="card card-body">
          <form onSubmit={onSubmitFunc}>
            <div className="form-group">
              <label htmlFor="searchField">Search a Title</label>
              <input
                type="text"
                className="form-control"
                id="searchField"
                placeholder="Enter part of a title"
                value={searchValue}
                onChange={event => onChangeFunc(event.target.value)}
                ref={node => this.input = node}
              />
              <small id="searchFieldHelp" className="form-text text-muted">
                Searching for remote news
              </small>
            </div>
            <button
              className="btn btn-primary"
              type="submit"
              data-toggle="collapse"
              data-target="#collapseExample"
              aria-expanded="false"
              aria-controls="collapseExample"
            >
              {children}
            </button>
          </form>
        </div>
        <br />
      </div>
    );
  }
} 

SearchField.propTypes = {
  onChangeFunc: PropTypes.func.isRequired,
  onSubmitFunc: PropTypes.func.isRequired,
  searchValue: PropTypes.string.isRequired,
  children: PropTypes.string
};

export default SearchField;
