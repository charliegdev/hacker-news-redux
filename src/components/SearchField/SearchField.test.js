import React from "React";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import SearchField from "./SearchField";

describe("SearchField", () => {
  const mockFunc = () => "mockFunc";
  const mockSearchField = <SearchField
    onChangeFunc={mockFunc}
    onSubmitFunc={mockFunc}
    searchValue="mock search value"
  >
    Search for Something
  </SearchField>;
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(mockSearchField, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(mockSearchField);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
