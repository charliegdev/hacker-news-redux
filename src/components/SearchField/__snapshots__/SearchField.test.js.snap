// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField has a valid snapshot 1`] = `
<div>
  <div
    className="ui segment"
  >
    <form
      onSubmit={[Function]}
    >
      <h3
        className="ui header"
      >
        Search for Something
      </h3>
      <div
        className="ui search"
      >
        <div
          className="ui icon input"
        >
          <input
            className="prompt"
            onChange={[Function]}
            placeholder="Search for a title..."
            type="text"
            value="mock search value"
          />
          <i
            className="search icon"
          />
        </div>
        <div
          className="results"
        />
      </div>
    </form>
  </div>
  <br />
</div>
`;
